### PR TITLE
make port a required field

### PR DESCRIPTION
**Pull request by hbagdi**
See #414

### CODE REVIEW

**Comment by robscott** [comment]
Thanks!

/approve

**Comment by robscott** [comment]
@hbagdi It looks like this will require some updates to our examples to actually specify ports.

**Comment by hbagdi** [comment]
> @hbagdi It looks like this will require some updates to our examples to actually specify ports.

Ah, the examples as always. We really need CI to catch that. Waiting for #346.
Thanks for catching that.

/hold

**Comment by hbagdi** [comment]
Updated.
/hold cancel

**Comment by jpeach** [comment]
Making the port required is a good idea, but I don't think we should set a default.

Do arbitrary apps really listen on port 80 by default? I would be pretty surprised. Even if there was a predominance of ports 80 listeners, that requires running as root in the container (for the low-numbered port) and we should prefer to not encourage that :)

**Comment by howardjohn** [comment]
> Do arbitrary apps really listen on port 80 by default? I would be pretty surprised. Even if there was a predominance of ports 80 listeners, that requires running as root in the container (for the low-numbered port) and we should prefer to not encourage that :)

I cannot give any real numbers other than general feeling but it seems `servicePort=80,targetPort>1024` is pretty common from what I have seen, so I don't think this encourages root?

**Comment by robscott** [comment]
Yeah I think it's good to keep in mind that this is the port the Service is listening on, not the container itself. It's probably fairly likely that an HTTP service would listen on 80 and then target some other port that the containers were listening on.

**Comment by jpeach** [comment]
> Yeah I think it's good to keep in mind that this is the port the Service is listening on, not the container itself.

Good point, I had forgotten that 👍 

>  It's probably fairly likely that an HTTP service would listen on 80 and then target some other port that the containers were listening on.

Maybe. I'm still unconvinced that this is an improvement. If the default is wrong you'll have to debug it (high cost), and any benefit is very small.

**Comment by robscott** [comment]
>Maybe. I'm still unconvinced that this is an improvement. If the default is wrong you'll have to debug it (high cost), and any benefit is very small.

👍 that's fair. Always easier to add a default in the future than to remove one. I'm fine with leaving that out of this change.

**Comment by hbagdi** [comment]
I agree with the thoughts here. Updated to remove the defaulting logic.

File: apis/v1alpha1/httproute_types.go [hunk 1]
```diff
@@ -535,24 +520,11 @@ type HTTPRouteForwardTo struct {
...
-	// should be updated with a condition that describes the error
-	// more specifically.
-	//
 	// Support: Core
 	//
-	// +optional
 	// +kubebuilder:validation:Minimum=1
 	// +kubebuilder:validation:Maximum=65535
-	Port *int32 `json:"port,omitempty"`
+	Port int32 `json:"port,omitempty"`
```
Comment (jpeach) [inline apis/v1alpha1/httproute_types.go:527]:
```suggestion
	Port int32 `json:"port"`
```

Comment (hbagdi) [reply apis/v1alpha1/httproute_types.go]: Updated.

**Comment by robscott** [comment]
Thanks!

/lgtm

**Pull request merged**